### PR TITLE
Feature#30 register user

### LINKED ISSUE #30: Register User
# REQUIREMENTS
As a user I want to see 
- [x] Phone number
- [x] Email
- [x] Name of store owner
- [x] Store address (street, number, colony, municipality, Z.P.)
- [x] Type of business
- [x] Kind of delivery (pick up in store, send to address)
- [x] Days when it can be send to address
- [x] Working Hours

- [x] Checkbox for "Politics and Terms"
- [x] Checkbox for "I want to receive information from Tribo"

- [x] Register button

# DESIGN
image

### PR DESCRIPTION
**Pull request by danielasotomayor99**
Register view for users was created, see the ticket #30 
 -  Inputs and submit button were created.
 -  Styles were added as in figma's dessign.
 -  If inputs are empty its styles change to warning.
 - Message of mail validation: if mail already exists in db user can not to registrer ass a new one and a message appears.
 - Responsive views for mobile, tablet and desktop.

  ## Screenshots 
Before: 
before
After:
image

### CODE REVIEW

**Review by KarinaDell — DISMISSED** [review]

**Review by EduJimenez96 — DISMISSED** [review]

File: db/migrate/20210223062251_change_index_from_name_to_email.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class AddUniquenessEmailColumnInUsers < ActiveRecord::Migration[6.0]
```
Comment (YilKanda) [inline db/migrate/20210223062251_change_index_from_name_to_email.rb]: At the time of migrations it shows this error:

imagen

You need change the name of migration to:

`20210223062251_add_uniqueness_email_column_in_users.rb`

File: app/javascript/packs/input.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+let i = 0;
+document.addEventListener('DOMContentLoaded', function () {
```
Comment (YilKanda) [inline app/javascript/packs/input.js:2]: It is suggested to replace 'DOMContentLoaded' with 'turbolinks:load'

File: app/javascript/packs/input.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  const elements = document.getElementsByTagName('INPUT');
+  console.log(elements[1]);
+  while (i < elements.length) {
```
Comment (YilKanda) [inline app/javascript/packs/input.js:5]: WDYT about using a more modern notation like `elements.forEach`?

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/javascript/packs/input.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    elements[i].oninvalid = function (e) {
+      console.log(e);
+      console.log(e.target);
```
Comment (JacquelineFM) [inline app/javascript/packs/input.js:8]: Can you remove these logs? If not then it's okay.

File: app/javascript/packs/input.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+let i = 0;
+document.addEventListener('DOMContentLoaded', function () {
+  const elements = document.getElementsByTagName('INPUT');
+  console.log(elements[1]);
```
Comment (JacquelineFM) [inline app/javascript/packs/input.js:4]: Can you remove this log?

**Review by JacquelineFM — CHANGES_REQUESTED** [review]

**Review by EduJimenez96 — APPROVED** [review]

**Pull request merged**